onnxruntime/core/platform/env.h

```diff
@@ -108,8 +104,21 @@ class Env {
 
   virtual int GetNumCpuCores() const = 0;
 
-  // This function doesn't support systems with more than 64 logical processors
-  virtual std::vector<size_t> GetThreadAffinityMasks() const = 0;
+  // Return default threadpool size, and set default affinity vector.
+  // Note - affinity.size() does not necessarily amount to the threadpool size,
+  // e.g., for windows, affinities vector could be 0,1,0,2, which stands for ((0,1),(0,2)),
+  // the first pair (0,1) will be applied to a thread, the second pair (0,2) will be applied to another,
+  // so the returned value will be 2.
+  // For POSIX, the implementation is entirely different, affinity.size() will be rightly the threadpool size.
```

Comment:
>For POSIX, the implementation is entirely different, affinity.size() will be rightly the threadpool size. [](http://example.com/codeflow?start=5&length=105)

This is a platform independent interface. People should expect it to have the same spec for all platforms.

Reply:
For main branch, it can unified since per you latest change we have a vector of vector<size>.